sklearn/inspection/_permutation_importance.py

```diff
@@ -262,7 +269,12 @@ def permutation_importance(
     >>> result.importances_std
     array([0.2211..., 0.       , 0.       ])
     """
-    if not hasattr(X, "iloc"):
+    if _is_pandas_df(X):
+        adapter = ADAPTERS_MANAGER.adapters["pandas"]
+    elif _is_polars_df(X):
+        adapter = ADAPTERS_MANAGER.adapters["polars"]
+    else:
+        adapter = None
```

Comment:
I think that we should use `_get_adapter_from_container`

Reply:
basically something like:

```python
if _is_pandas_df(X) or _is_polars_df(X):
    adapter = _get_adapter_from_container(X)
else:
    adapter = None

Reply:
`_get_adapter_from_container` seems to be defined in `preprocessing/_function_transformer.py` rather than in `utils/_set_output.py`. Should I import from there, or should I move it to under `utils/_set_output.py`?

Reply:
Actually, we have a similar pull-request (https://github.com/scikit-learn/scikit-learn/pull/28718) where we move the function to `_set_output` (with the tests). We can do the same changes and depending which PR go in first, we will have to solve the conflicts ;)